Assets/Mirage/Runtime/Sockets/Udp/UdpSocketFactory.cs

```diff
@@ -70,18 +69,25 @@ void ThrowIfNotSupported()
 
     public class UdpSocket : ISocket
     {
-        readonly Socket socket;
+        Socket socket;
+        IPEndPoint AnyEndpoint;
 
         public UdpSocket()
         {
-            // todo do we need to use AddressFamily from endpoint?
-            socket = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);
-            socket.Blocking = false;
-            socket.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.ReuseAddress, true);
         }
 
         public void Bind(EndPoint endPoint)
         {
+            AnyEndpoint = endPoint as IPEndPoint;
```

Comment:
`socket.ReceiveFrom` needs an Endpoint to be passed in, Some examples I saw just pass in `Any`. I set that Endpoint to the the value given to Bind/Connect instead of creating a new one, I think this way it will ensure it is the correct address family